<a href="https://colab.research.google.com/github/alimhdan/Home-Credit-Indonesia/blob/main/Home_Credit_Indonesia_Market_Campaign.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

**PERMASALAHAN**

Tim Produk, dibantu oleh Tim Marketing, saat ini sedang meluncurkan dua produk/jasa terbaru, dan salah satunya merupakan jasa yang digunakan untuk menunjang penjualan produk yang sudah ada, produk PINJAMAN DANA MULTIGUNA. Oleh karena itu, mereka sedang membutuhkan Tim Data untuk melakukan beberapa analisa yang sangat mendesak untuk membantu mereka meningkatkan peluang cross-selling produk/jasa terbaru mereka untuk tahap awal (A/B testing). Sayangnya, saat ini tidak ada BI yang tersedia di Business Intelligence Center. Oleh karena itu, seorang Data Scientist di tim CRM Analytics dan anda sebagai Data Scientist Intern akan ditugaskan menyelesaikan permasalahan tersebut. Ini adalah peluang anda untuk menunjukkan kemampuan anda dan juga sebagai jalan untuk anda membiasakan diri dengan apa yang Data Scientist di Home Credit kerjakan.

Mereka ingin mengetahui pada grup umur saat mendaftar yang mana mayoritas pelanggan tersegmentasi jika :

- "Late Ten" : Age<20

- "Twenty" : 20<=Age<30

- "Thirty" : 30<=Age<40

- "Fourty" : 40<=Age<50

- "Fifty" : 50<=Age<60

- "Above_Sixty" : 60<=Age

Sebagai catatan, tim marketing hanya membutuhkan data pelanggan yang tidak memiliki kesulitan pembayaran (karena ini terkait dengan kemampuan mereka dalam membeli produk) dan tinggal di wilayah yang diberi rating 3 (REGION_RATING_CLIENT_W_CITY) serta melakukan transaksi pada tanggal 1 Januari 2017 - 31 Desember 2018.

Data yang dibutuhkan untuk analisis ini akan disediakan oleh Tim Marketing dan dapat diunduh pada bagian Sumber Daya yang Dapat Digunakan.

Tentukan 3 urutan grup usia yang paling dominan!


---




**SOLUSI**

In [ ]:
#memanggil data yang diinginkan

import pandas as pd
data=pd.read_csv("/content/drive/MyDrive/Colab Notebooks/Marketing_Campaign_Data_A-21492758-f47a-4913-971e-68c9a721311a.csv")
data.head()

Unnamed: 0  TARGET NAME_CONTRACT_TYPE CODE_GENDER FLAG_OWN_CAR  \
0       58785       0         Cash loans           F            Y   
1      166116       0         Cash loans           F            N   
2      170111       0         Cash loans           F            Y   
3      184247       0         Cash loans           F            N   
4       75886       0         Cash loans           M            Y   

  FLAG_OWN_REALTY  AMT_INCOME_TOTAL  AMT_CREDIT  AMT_ANNUITY  AMT_GOODS_PRICE  \
0               Y          157500.0    225000.0      11781.0         225000.0   
1               Y          180000.0    508495.5      21672.0         454500.0   
2               Y          360000.0    639396.0      47934.0         571500.0   
3               Y          162000.0    254700.0      14751.0         225000.0   
4               Y          225000.0    787131.0      42066.0         679500.0   

             NAME_EDUCATION_TYPE  NAME_HOUSING_TYPE  \
0  Secondary / secondary special  House / apartment   
1  Secondary / secondary special  House / apartment   
2              Incomplete higher  House / apartment   
3               Higher education  House / apartment   
4               Higher education  House / apartment   

   REGION_POPULATION_RELATIVE  REGION_RATING_CLIENT_W_CITY  \
0                    0.014520                            2   
1                    0.031329                            2   
2                    0.032561                            1   
3                    0.010147                            2   
4                    0.025164                            2   

         APPL_DATETIME  DATE_BIRTH  
0  2016-06-11 17:28:54  17-07-1959  
1  2015-05-22 08:42:03  23-06-1951  
2  2016-11-26 20:54:07  07-06-1979  
3  2017-02-20 09:22:06  20-07-1958  
4  2017-02-28 09:38:45  05-08-1985

In [ ]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 30722 entries, 0 to 30721
Data columns (total 16 columns):
 #   Column                       Non-Null Count  Dtype  
---  ------                       --------------  -----  
 0   Unnamed: 0                   30722 non-null  int64  
 1   TARGET                       30722 non-null  int64  
 2   NAME_CONTRACT_TYPE           30722 non-null  object 
 3   CODE_GENDER                  30722 non-null  object 
 4   FLAG_OWN_CAR                 30722 non-null  object 
 5   FLAG_OWN_REALTY              30722 non-null  object 
 6   AMT_INCOME_TOTAL             30722 non-null  float64
 7   AMT_CREDIT                   30722 non-null  float64
 8   AMT_ANNUITY                  30722 non-null  float64
 9   AMT_GOODS_PRICE              30722 non-null  float64
 10  NAME_EDUCATION_TYPE          30722 non-null  object 
 11  NAME_HOUSING_TYPE            30722 non-null  object 
 12  REGION_POPULATION_RELATIVE   30722 non-null  float64
 13  REGION_RATING_CL

In [ ]:
#mengganti tipe data

import datetime as dt

data['APPL_DATETIME']=pd.to_datetime(data['APPL_DATETIME']).dt.date
data['APPL_DATETIME']

0        2016-06-11
1        2015-05-22
2        2016-11-26
3        2017-02-20
4        2017-02-28
            ...    
30717    2017-09-23
30718    2018-01-08
30719    2018-04-29
30720    2017-11-05
30721    2018-11-30
Name: APPL_DATETIME, Length: 30722, dtype: object

In [ ]:
data['DATE_BIRTH']=pd.to_datetime(data['DATE_BIRTH']).dt.date
data['DATE_BIRTH']

0        1959-07-17
1        1951-06-23
2        1979-07-06
3        1958-07-20
4        1985-05-08
            ...    
30717    1953-06-02
30718    1984-06-14
30719    1980-09-10
30720    1964-12-13
30721    1961-03-30
Name: DATE_BIRTH, Length: 30722, dtype: object

In [ ]:
#menghitung umur

AGE=data['APPL_DATETIME']-data['DATE_BIRTH']
AGE.head()

0   20784 days
1   23344 days
2   13658 days
3   21400 days
4   11619 days
dtype: timedelta64[ns]

In [ ]:
#Konversi ke bentuk integer

import numpy as np

AGE_DAYS = (AGE/np.timedelta64(1, 'D')).astype(int)
AGE_DAYS

0        20784
1        23344
2        13658
3        21400
4        11619
         ...  
30717    23489
30718    12261
30719    13745
30720    19320
30721    21064
Length: 30722, dtype: int64

In [ ]:
#menghitung umur dalam Tahun

AGE_YEARS = AGE_DAYS/365
AGE_YEARS

0        56.942466
1        63.956164
2        37.419178
3        58.630137
4        31.832877
           ...    
30717    64.353425
30718    33.591781
30719    37.657534
30720    52.931507
30721    57.709589
Length: 30722, dtype: float64

Segmentasi grup umur pelanggan akan dikategorikan berdasarkan AGE_YEARS:

"Late Ten" : Age<20

"Twenty" : 20<=Age<30

"Thirty" : 30<=Age<40

"Fourty" : 40<=Age<50

"Fifty" : 50<=Age<60

"Above_Sixty" : 60<=Age

In [ ]:
#Membentuk segmentasi grup umur

def age(Ayer):
    if (Ayer<20):
     return "Late Ten"
    if (Ayer >=20 and Ayer < 30):
     return "Twenty"
    if (Ayer >=30 and Ayer < 40):
     return "Thirty"
    if (Ayer >=40 and Ayer < 50):
     return "Fourty"
    if (Ayer >=50 and Ayer < 60):
     return "Fifty"
    if (Ayer >= 60):
     return "Above Sixty"

print([age(AG) for AG in AGE_YEARS])

['Fifty', 'Above Sixty', 'Thirty', 'Fifty', 'Thirty', 'Twenty', 'Fourty', 'Above Sixty', 'Above Sixty', 'Twenty', 'Twenty', 'Above Sixty', 'Fourty', 'Fifty', 'Above Sixty', 'Thirty', 'Fourty', 'Twenty', 'Fifty', 'Twenty', 'Above Sixty', 'Above Sixty', 'Fourty', 'Above Sixty', 'Fourty', 'Twenty', 'Twenty', 'Fourty', 'Fifty', 'Above Sixty', 'Above Sixty', 'Thirty', 'Thirty', 'Fifty', 'Above Sixty', 'Fifty', 'Above Sixty', 'Fifty', 'Fourty', 'Fifty', 'Twenty', 'Fourty', 'Twenty', 'Twenty', 'Fourty', 'Fifty', 'Above Sixty', 'Thirty', 'Above Sixty', 'Above Sixty', 'Fifty', 'Fifty', 'Fifty', 'Thirty', 'Thirty', 'Thirty', 'Above Sixty', 'Fifty', 'Fifty', 'Thirty', 'Fifty', 'Thirty', 'Above Sixty', 'Twenty', 'Fifty', 'Twenty', 'Twenty', 'Fifty', 'Fourty', 'Thirty', 'Twenty', 'Twenty', 'Fourty', 'Above Sixty', 'Twenty', 'Twenty', 'Fifty', 'Above Sixty', 'Above Sixty', 'Thirty', 'Fourty', 'Fifty', 'Fifty', 'Fourty', 'Fifty', 'Thirty', 'Fifty', 'Thirty', 'Fourty', 'Above Sixty', 'Fifty', 'Fifty',

In [ ]:
#menggabungkan variabel ke dalam dataset

data_fiks=data.assign(AGE=AGE_YEARS, AGE_CATEGORY=[age(AG) for AG in AGE_YEARS])
data_fiks.head()

Unnamed: 0  TARGET NAME_CONTRACT_TYPE CODE_GENDER FLAG_OWN_CAR  \
0       58785       0         Cash loans           F            Y   
1      166116       0         Cash loans           F            N   
2      170111       0         Cash loans           F            Y   
3      184247       0         Cash loans           F            N   
4       75886       0         Cash loans           M            Y   

  FLAG_OWN_REALTY  AMT_INCOME_TOTAL  AMT_CREDIT  AMT_ANNUITY  AMT_GOODS_PRICE  \
0               Y          157500.0    225000.0      11781.0         225000.0   
1               Y          180000.0    508495.5      21672.0         454500.0   
2               Y          360000.0    639396.0      47934.0         571500.0   
3               Y          162000.0    254700.0      14751.0         225000.0   
4               Y          225000.0    787131.0      42066.0         679500.0   

             NAME_EDUCATION_TYPE  NAME_HOUSING_TYPE  \
0  Secondary / secondary special  House / apartment   
1  Secondary / secondary special  House / apartment   
2              Incomplete higher  House / apartment   
3               Higher education  House / apartment   
4               Higher education  House / apartment   

   REGION_POPULATION_RELATIVE  REGION_RATING_CLIENT_W_CITY APPL_DATETIME  \
0                    0.014520                            2    2016-06-11   
1                    0.031329                            2    2015-05-22   
2                    0.032561                            1    2016-11-26   
3                    0.010147                            2    2017-02-20   
4                    0.025164                            2    2017-02-28   

   DATE_BIRTH        AGE AGE_CATEGORY  
0  1959-07-17  56.942466        Fifty  
1  1951-06-23  63.956164  Above Sixty  
2  1979-07-06  37.419178       Thirty  
3  1958-07-20  58.630137        Fifty  
4  1985-05-08  31.832877       Thirty

Data yang akan digunakan adalah data yang memnuhi kriteria sebagai berikut:
* Data pelanggan yang tidak memiliki kesulitan pembayaran (karena ini terkait 
dengan kemampuan mereka dalam membeli produk) ==> TARGET == 0

* Tinggal di wilayah yang diberi rating 3 (REGION_RATING_CLIENT_W_CITY) ==> REGION_RATING_CLIENT_W_CITY == 3

* Melakukan transaksi pada tanggal 1 Januari 2017 - 31 Desember 2018 ==> APPL_DATETIME pada tanggal 2017-01-01 sampai 2018-12-31

In [ ]:
#melakukan filtering data yang dibutuhkan

data_fiks1=data_fiks[(data_fiks["TARGET"] == 0) & 
                    (data_fiks["REGION_RATING_CLIENT_W_CITY"] == 3)]
data_fiks1.head()

Unnamed: 0  TARGET NAME_CONTRACT_TYPE CODE_GENDER FLAG_OWN_CAR  \
17       58409       0         Cash loans           F            N   
21       95421       0         Cash loans           F            N   
26      180539       0         Cash loans           F            N   
31      162472       0         Cash loans           M            Y   
36      188535       0    Revolving loans           F            N   

   FLAG_OWN_REALTY  AMT_INCOME_TOTAL  AMT_CREDIT  AMT_ANNUITY  \
17               Y           90000.0    585000.0      21141.0   
21               Y          135000.0    648000.0      33214.5   
26               Y          225000.0    545040.0      25537.5   
31               Y          371250.0    545040.0      20677.5   
36               Y           67500.0    135000.0       6750.0   

    AMT_GOODS_PRICE            NAME_EDUCATION_TYPE  NAME_HOUSING_TYPE  \
17         585000.0  Secondary / secondary special  House / apartment   
21         648000.0  Secondary / secondary special  House / apartment   
26         450000.0  Secondary / secondary special  House / apartment   
31         450000.0  Secondary / secondary special  House / apartment   
36         135000.0  Secondary / secondary special  House / apartment   

    REGION_POPULATION_RELATIVE  REGION_RATING_CLIENT_W_CITY APPL_DATETIME  \
17                    0.018209                            3    2016-10-10   
21                    0.018209                            3    2017-04-22   
26                    0.010500                            3    2018-05-26   
31                    0.018209                            3    2016-12-25   
36                    0.018029                            3    2015-11-24   

    DATE_BIRTH        AGE AGE_CATEGORY  
17  1995-01-02  21.786301       Twenty  
21  1949-12-13  67.402740  Above Sixty  
26  1993-07-28  24.843836       Twenty  
31  1983-10-17  33.213699       Thirty  
36  1955-10-12  60.158904  Above Sixty

In [ ]:
import datetime

start=datetime.datetime.strptime('2017-01-01', "%Y-%m-%d").date()
end=datetime.datetime.strptime('2018-12-31', "%Y-%m-%d").date()

data_fiks2=data_fiks1[(data_fiks1['APPL_DATETIME'] >= start) & 
                      (data_fiks1['APPL_DATETIME'] <= end)]
data_fiks2.sort_values('APPL_DATETIME')

Unnamed: 0  TARGET NAME_CONTRACT_TYPE CODE_GENDER FLAG_OWN_CAR  \
26868      170853       0    Revolving loans           M            N   
9898        69879       0         Cash loans           M            N   
9399       117071       0         Cash loans           F            Y   
26602      177007       0         Cash loans           M            Y   
22089       77260       0         Cash loans           F            N   
...           ...     ...                ...         ...          ...   
3919       160761       0         Cash loans           M            N   
19932      182288       0         Cash loans           F            N   
25171      132798       0         Cash loans           F            N   
19216       91926       0         Cash loans           M            Y   
13428      169276       0         Cash loans           F            N   

      FLAG_OWN_REALTY  AMT_INCOME_TOTAL  AMT_CREDIT  AMT_ANNUITY  \
26868               Y           90000.0    180000.0       9000.0   
9898                Y          270000.0   2013840.0      55507.5   
9399                Y           85500.0    675000.0      21775.5   
26602               N          180000.0    755190.0      36459.0   
22089               N          112500.0    355500.0      18891.0   
...               ...               ...         ...          ...   
3919                Y          105750.0   1133748.0      33277.5   
19932               Y          112500.0    170640.0       9922.5   
25171               Y           90000.0    565380.0      28998.0   
19216               N          166500.0    566055.0      20925.0   
13428               Y          112500.0    369000.0      14040.0   

       AMT_GOODS_PRICE            NAME_EDUCATION_TYPE  NAME_HOUSING_TYPE  \
26868         180000.0               Higher education   Office apartment   
9898         1800000.0  Secondary / secondary special  House / apartment   
9399          675000.0               Higher education  House / apartment   
26602         675000.0  Secondary / secondary special  House / apartment   
22089         355500.0  Secondary / secondary special  House / apartment   
...                ...                            ...                ...   
3919          990000.0  Secondary / secondary special  House / apartment   
19932         135000.0               Higher education  House / apartment   
25171         423000.0  Secondary / secondary special  House / apartment   
19216         472500.0  Secondary / secondary special  House / apartment   
13428         369000.0               Higher education  House / apartment   

       REGION_POPULATION_RELATIVE  REGION_RATING_CLIENT_W_CITY APPL_DATETIME  \
26868                    0.020713                            3    2017-01-01   
9898                     0.018029                            3    2017-01-01   
9399                     0.002042                            3    2017-01-01   
26602                    0.020246                            3    2017-01-02   
22089                    0.018209                            3    2017-01-03   
...                           ...                          ...           ...   
3919                     0.006852                            3    2018-12-27   
19932                    0.002134                            3    2018-12-28   
25171                    0.006296                            3    2018-12-29   
19216                    0.018029                            3    2018-12-29   
13428                    0.018209                            3    2018-12-30   

       DATE_BIRTH        AGE AGE_CATEGORY  
26868  1995-12-18  21.054795       Twenty  
9898   1950-09-27  66.309589  Above Sixty  
9399   1965-11-19  51.153425        Fifty  
26602  1981-08-03  35.441096       Thirty  
22089  1963-10-03  53.290411        Fifty  
...           ...        ...          ...  
3919   1955-06-08  63.597260  Above Sixty  
19932  1956-03-13  62.835616  Above Sixty  
25171  1985-11-10  33.156164       Thirty  

Setelah data bersih dan sudah sesuai dengan kualifikasi, langkah terakhir adalah menghitung jumlah segmen grup umur yang dominan menggunakan syntax 

```
groupby
```



In [ ]:
#menghitung jumlah AGE_CATEGORY

data_fiks2.groupby('AGE_CATEGORY')['AGE_CATEGORY'].count().sort_values(ascending=False)

AGE_CATEGORY
Fifty          575
Above Sixty    495
Twenty         364
Thirty         300
Fourty         203
Name: AGE_CATEGORY, dtype: int64

Berdasarkan output tersebut dapat dilihat bahwa jumlah grup umur yang paling dominan adalah Fifty yaitu sejumlah 575 pelanggan, kemudian Above Sixty yaitu sejumlah 495 pelanggan, dan Twenty sejumlah 364 pelanggan.